### DISCUSSION TITLE
Supabase JS client doesn't delete data

### DISCUSSION THREAD

**Discussion by rdmchr** [supabase/supabase#1617] [Questions] [answered]
I'm using the `@supabase/supabase-js` client for my project and I'm trying to delete some data in the database. I use this class to create the client:
```JS
import { createClient } from '@supabase/supabase-js'

// Create a single supabase client for interacting with your database
const supabase = createClient(<URL>, <TOKEN>, {autoRefreshToken: true, persistSession: true})

const { auth, storage } = supabase;
const usersDatabase = supabase.from('users');
const memoriesDatabase = supabase.from('memories');
const tagsDatabase = supabase.from('tags');
const mappingDatabase = supabase.from('mem_to_tag');

export {
    supabase,
    auth,
    storage,
    usersDatabase,
    memoriesDatabase,
    tagsDatabase,
    mappingDatabase,
}
```
and I'm trying to delete some data using this code:
```JS
const result = tagsDatabase.delete().match({color: "#0000a0"});
console.log(result);
```
But the only output I see is:
`PostgrestFilterBuilder {url: URL, headers: {…}, schema: "public", _subscription: SupabaseRealtimeClient, _realtime: RealtimeClient, …}`
And the data is not being deleted. I have confirmed that the match part is correct and that there is a database entry with that color.
I also already tried deleting using a match that would target the id/primary key of the database entry but that yielded the same result.

**Comment by inian** [comment, marked as answer]
You are missing an await @rdmchr 
```
const result = await tagsDatabase.delete().match({color: "#0000a0"});

```

**Reply by rdmchr** [reply]
Oh. That was a stupid mistake. Thank you very much :)
